### PR TITLE
Updating Postgres & Java versions in sysadmin docs

### PR DESCRIPTION
**Pull request by hflynn**
See https://trello.com/c/rw0fDfm0/303-update-postgres-minimum-version-from-8-4-to-9-3 and https://trello.com/c/JhyR5Z93/249-deprecate-versions-java-1-6-etc

Updated version requirements based on @rleigh-dundee epic page, which still needs adding to the sysadmin what's new page once Simon's changes are merged.

### CODE REVIEW

File: omero/sysadmins/unix/server-installation.txt [hunk 1]
```diff
@@ -136,9 +136,8 @@ If possible, install one of the following packages:
 +-----------+---------------------------+
 
 OMERO works with the OpenJDK JRE provided by most systems, or with
-Oracle Java. Version 6 or later is required. Version 7 or later is
-recommended (version 6 is end of line and no longer has updates or
-security support). **OpenJDK version 8 is not currently supported**.
+Oracle Java. Version 7 or later is
+recommended. **OpenJDK version 8 is not currently supported**.
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.txt]: Isn't OpenJDK 8 supported?

File: omero/sysadmins/unix/server-installation.txt [hunk 2]
```diff
@@ -175,8 +174,8 @@ Check you have Python (and check its version) by running:
     $ python --version
     Python 2.7.5+
 
-.. note:: OMERO does not currently support Python 3; you must use 2.6 or
-   a later 2.x release.
+.. note:: OMERO does not currently support Python 3; you must use 2.6 or a
+    later 2.x release.
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.txt]: I'd be tempted to get rid of the `x` and say Python 2.6 or 2.7 for clarity, since they're the only two versions.

File: omero/sysadmins/unix/server-installation.txt [hunk 3]
```diff
@@ -270,19 +269,8 @@ If possible, install one of the following packages:
...
-| Debian | 7.0          | 3.4               | 4.4.x, 5.x.x  |
-+--------+--------------+-------------------+---------------+
-| Ubuntu | 14.04 (LTS)  | 3.5               | 4.4.x, 5.x.x  |
-|        +--------------+-------------------+---------------+
-|        | 13.10        | 3.4               | 4.4.x, 5.x.x  |
-|        +--------------+-------------------+---------------+
-|        | 12.04 (LTS)  | 3.4               | 4.4.x, 5.x.x  |
-+--------+--------------+-------------------+---------------+
+and `Ubuntu <https://wiki.ubuntu.com/Releases>`_ are shown in the
+:ref:`ice-section` of the :doc:`/sysadmins/version-requirements` page.
```
Comment (rleigh-dundee) [inline omero/sysadmins/unix/server-installation.txt]: Maybe use `ice-requirements` or `ice-version-requirements` as the cross ref indentifer; no need to use `-section` since the cross-refs are to sections so that's already implied.

File: omero/sysadmins/unix/server-installation.txt [hunk 4]
```diff
@@ -251,7 +250,7 @@ Ice
 .. note::
 
     OMERO |version| supports 3.5 and 3.4 on UNIX and UNIX-like
-    platforms. You must install the correct version of OMERO.server
+    platforms.You must install the correct version of OMERO.server
```
Comment (manics) [inline omero/sysadmins/unix/server-installation.txt]: Mysterious disappearing space?

**Comment by manics** [comment]
There's a few mentions of Postgres 9.1 (run `git grep 9\\.1 -- \*.txt`).

File: omero/sysadmins/unix/server-installation.txt (hunk 1 above)
Comment (rleigh-dundee) [inline omero/sysadmins/unix/server-installation.txt]: Some versions e.g. in Ubuntu 14.10 have completely broken jpeg/image loading support.  When you run Insight all you see is a bunch of black squares for every image.  On FreeBSD it does work, with a different OpenJDK release, so it's not that we don't support it, it's that some versions in common use are unusable at present.

**Comment by hflynn** [comment]
I'll fix the 9.1 stuff tomorrow but that's the other comments fixed.

File: omero/sysadmins/unix/server-installation.txt (hunk 1 above)
Comment (hflynn) [inline omero/sysadmins/unix/server-installation.txt]: do you want me to update that then or just leave it?

Comment (rleigh-dundee) [inline omero/sysadmins/unix/server-installation.txt]: Either is fine.  I stated it this way because we don't know exactly which OpenJDK8 versions/platforms are affected, so it was safer to say it was unsupported, i.e. use at own risk.

File: omero/sysadmins/windows/server-installation.txt [hunk 5]
```diff
@@ -151,12 +152,12 @@ The following are optional depending on what services you require:
 
 .. [2] Allows larger volumes to be viewed in the :omero_plone:`Volume Viewer <volume-viewer-in-omero.web>`.
 
-PostgreSQL (8.4 or higher)
+PostgreSQL (9.2 or higher)
 ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 PostgreSQL has to be installed and configured with PL/pgSQL and to
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Remove `with PL/pgSQL and`l this is now enabled by default

File: omero/sysadmins/windows/server-installation.txt [hunk 6]
```diff
@@ -151,12 +152,12 @@ The following are optional depending on what services you require:
...
-accept TCP connections. PostgreSQL 8.3 and earlier are not supported.
-9.2 or later is recommended.
+accept TCP connections. PostgreSQL 9.1 and earlier are not supported.
+9.3 or later is recommended.
 
 The *One click installer* can be found on the `PostgreSQL Windows
 download page <http://www.postgresql.org/download/windows>`_. You
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: The comment about the password here is out of place, what's it got to do with the installer?

File: omero/sysadmins/windows/server-installation.txt [hunk 7]
```diff
@@ -166,81 +167,30 @@ will need the postgres user's password later in the install.
    this document; other PostgreSQL installation environments are
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Maybe delete the other possibilities here and just leave as "PostgreSQL must be installed as a service".

File: omero/sysadmins/windows/server-installation.txt [hunk 8]
```diff
@@ -166,81 +167,30 @@ will need the postgres user's password later in the install.
...
    supported but are outside the scope of this document.
 
-#. Run the downloaded installer:
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Steps 1-3 are really a single step and can be consolidated.

File: omero/sysadmins/windows/server-installation.txt [hunk 9]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
 
 -  Create a non-superuser database user (make sure to note down the name
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: The sentence at the start of this section can be reworded to remove the waffle.  "Probably the most important step towards having a working server system is having a properly configured database."

Let's remove it entirely or reword to e.g. "Create a database and user for OMERO as follows:"

File: omero/sysadmins/windows/server-installation.txt [hunk 10]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: New login role
-
-         New login role
-
-   #. Create a new role and record the name and password. You will need to 
-      configure OMERO to use your username and password by setting the 
+   #. Create a new role and record the name and password. You will need to
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Maybe reword.  They need to provide a username and password here before noting them down (the detail from the screenshot), and it needs stating here since there's nothing to record except.

File: omero/sysadmins/windows/server-installation.txt [hunk 11]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
 
-      .. figure:: /images/install-windows-screenshots/pgadmin-07newdatabase-context.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: New database
-
-         New database
-
    #. Create a new database with the :guilabel:`Name` ``omero_database`` and
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Emphasise not to use these example names?

File: omero/sysadmins/windows/server-installation.txt [hunk 12]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
 
-      .. figure:: /images/install-windows-screenshots/pgadmin-08newdatabase-name.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: New database name
-
-         New database name
-
 -  Confirm PL/pgSQL language support in your newly created database
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: This last section can be removed entirely since it's done by default.  All the way down to and including "Finally, add the plpgsql language, accepting all defaults."

**Comment by hflynn** [comment]
@rleigh-dundee do you want to check the new postgres workflow for me today and I'll fix this up in one go once we know what else needs updating? Thanks.

File: omero/sysadmins/windows/server-installation.txt [hunk 13]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-      .. figure::  /images/install-windows-screenshots/pgadmin-03connect.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: Enter password
-
-         Enter password
-
    #. Right-click on :guilabel:`Login Roles` and select :guilabel:`New Login
       Role...`
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Here add:

On the ``Properties`` tab enter the name of the OMERO database user (example: ``omero``) as the ``Role name``.  On the ``Definition`` tab, enter the password for the user twice.  Press ``OK``.

(The user name example could be omitted if covered above.)  This then follows on to the "record the name and password" bit.

File: omero/sysadmins/windows/server-installation.txt [hunk 14]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-
-      .. figure::  /images/install-windows-screenshots/pgadmin-03connect.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: Enter password
-
-         Enter password
-
    #. Right-click on :guilabel:`Login Roles` and select :guilabel:`New Login
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Create a new database user for OMERO.

File: omero/sysadmins/windows/server-installation.txt [hunk 15]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-
-   .. figure:: /images/install-windows-screenshots/pgadmin-06newrole-passwd.png
-      :figwidth: 65%
-      :width: 60%
-      :align: center
-      :alt: New role password
-
-      Setting password of new login role
 
 -  Create an ``omero_database`` database:
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Create a new database for OMERO.

Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Here add:

On the ``Properties`` tab enter the name of the OMERO database (example: ``omero``) as the ``Name`` and the role name of the OMERO database user created above as the ``Owner``.  Press ``OK``.

Note may need to add the default db name here from above if needed.

File: omero/sysadmins/windows/server-installation.txt [hunk 16]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-      .. figure:: /images/install-windows-screenshots/pgadmin-10viewlanguages.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: Enable display of installed languages
-
-         Enable display of installed languages
+      :guilabel:`Browser` tab and activate the *Languages* option
 
    #. Navigate back to your database, expand the database's tree view
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Delete

File: omero/sysadmins/windows/server-installation.txt [hunk 17]
```diff
@@ -328,120 +278,35 @@ is having a properly configured database.
...
-      .. figure:: /images/install-windows-screenshots/pgadmin-11installedlanguages.png
-         :figwidth: 65%
-         :width: 60%
-         :align: center
-         :alt: View installed languages
-
-         View installed languages
+      and finally expand the now available :guilabel:`Languages` item
 
    #. If the ``plpgsql`` language is missing, right-click on the
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Delete

File: omero/sysadmins/windows/server-installation.txt [hunk 18]
```diff
@@ -452,23 +317,6 @@ is having a properly configured database.
       the :guilabel:`New language...` option in the menu. Finally, add the
```
Comment (rleigh-dundee) [inline omero/sysadmins/windows/server-installation.txt]: Delete

**Comment by hflynn** [comment]
Should hopefully be sorted now

**Comment by rleigh-dundee** [comment]
@hflynn One last minor bit I didn't notice before.  It should be

- Open pgAdmin III
- Connect to the DB [first point of create user]
- Create database user
- Create database
```
i.e. just move the first point from create user above, since this also applies to both create steps.

Also, for "you will need to remember the password" (above), we should probably use "record" rather than "remember" as for the user name/password here.

**Comment by hflynn** [comment]
Done

**Comment by hflynn** [comment]
Have removed colon, adding a bit about performing other steps seems unnecessary given it's clearly a list of steps :)

**Comment by rleigh-dundee** [comment]
Thanks.  Looks good to merge.

Final very nitpicky point would be the use of full stops at the end of the bullet point sentences, but that can probably be left for another time if it's not matching the style elsewhere!

**Comment by hflynn** [comment]
They are actually ok here because the bullet points are full sentences and begin with capital letters ;)

**Comment by hflynn** [comment]
--no-rebase
